geo/src/algorithm/coords_iter.rs

```diff
@@ -95,476 +99,830 @@ pub trait CoordsIter {
     fn exterior_coords_iter(&self) -> Self::ExteriorIter<'_>;
 }
 
-// ┌──────────────────────────┐
-// │ Implementation for Point │
-// └──────────────────────────┘
+// Helper function for Polygon coordinate count
+fn polygon_coords_count<T, P>(polygon: &P) -> usize
+where
+    T: CoordNum,
+    P: PolygonTrait<T = T>,
+    // Bounds required by the function body (no lifetimes needed for count)
+    // P::RingType is implicitly bound via P: PolygonTrait
+{
+    let exterior_count = polygon
+        .exterior()
+        .map(|ls| LineStringTrait::num_coords(&ls))
+        .unwrap_or(0);
+    let interior_count = polygon
+        .interiors()
+        .map(|ls| LineStringTrait::num_coords(&ls))
+        .sum::<usize>();
+    exterior_count + interior_count
+}
+
+impl<G> CoordsIter for G
+where
+    G: GeometryTrait,
+    G::T: CoordNum,
+{
+    type T = G::T;
 
-impl<T: CoordNum> CoordsIter for Point<T> {
     type Iter<'a>
-        = iter::Once<Coord<T>>
+        = Iter<'a, G>
     where
-        T: 'a;
+        Self: 'a;
+
     type ExteriorIter<'a>
-        = Self::Iter<'a>
+        = ExteriorIter<'a, G>
     where
-        T: 'a;
-    type Scalar = T;
+        Self: Sized + 'a,
+        Self::T: 'a + CoordNum;
 
     fn coords_iter(&self) -> Self::Iter<'_> {
-        iter::once(self.0)
+        match self.as_type() {
+            GeometryType::Point(p) => Iter::Point(Some(PointTrait::coord(p).unwrap().to_coord())),
```

Comment:
This begs the question of how `geo` should handle empty points.

Reply:
I was intending to return an empty iterator here. Let me know if that doesn't sound right